app/uk/gov/hmrc/gform/core/parsers/ValueParser.scala

```diff
@@ -161,7 +163,15 @@ object ValueParser {
 
   private lazy val periodFun = "period(" ~ dateExpr ~ "," ~ dateExpr ~ ")"
 
-  lazy val contextField: Parser[Expr] = ("user" ~ "." ~ userField ^^ { (loc, _, _, userField) =>
+  private lazy val userEnrolmentFunc = "user" ~ "." ~ userFieldEnrolments ~ "." ~ "(count|[1-9][0-9]*)".r ^^ {
```

Comment:
Can we turn `(count|[1-9][0-9]*)` into `Parse[UserFuncCtx]`? Also we already have `intParser` which can be used here.

Reply:
Yes, definitely